libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.h

```diff
@@ -287,30 +258,52 @@ FOUNDATION_EXTERN NSString * const kSFLoginHostChangedNotificationUpdatedHostKey
 - (void)clearAllAccountState;
 
 /** Invoke this method to apply the specified credentials to the
- current user. If no user exists, a new one is created.
+ a user whose credentials match. If no user exists, a new one is created.
  This will post user update notification.
  @param credentials The credentials to apply
  */
-- (void)applyCredentials:(SFOAuthCredentials*)credentials;
+- (SFUserAccount *) applyCredentials:(SFOAuthCredentials*)credentials;
 
 /** Invoke this method to apply the specified id data to the
  current user. This will post user update notification.
  @param idData The ID data to apply
  */
 - (void)applyIdData:(SFIdentityData *)idData;
 
+/** Invoke this method to apply the specified id data to the
+  user. This will post user update notification.
+  @param idData The ID data to apply
+  @param user The SFUserAccount to apply this change to.
+ */
+- (void)applyIdData:(SFIdentityData *)idData forUser:(SFUserAccount *)user;
+
 /** This method will selectively update the custom attributes identity data for the current user.
  Other identity data will not be impacted.
  @param customAttributes The new custom attributes data to update in the identity data.
  */
 - (void)applyIdDataCustomAttributes:(NSDictionary *)customAttributes;
```

Comment:
Same comment as `applyIdData:`.  Seems like we should remove "current user" convenience front ends.